### PR TITLE
eset_protect: add HTTP header identifying the integration

### PR DESCRIPTION
**Pull request by eset-enterprise-integration**
## Proposed commit message

```
eset_protect: add HTTP header identifying the integration

The ESET Protect API will be requiring an HTTP header, "3rd-integration", to
identify integrations. Documentation for the header is not currently
available, but will be at the ESET Elastic integration documentation page[1]
in the future.

[1]https://help.eset.com/eset_connect/en-US/elastic.html
```

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [x] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices) 

## How to test this PR locally
We will see changes in our monitoring after release

## Related issues
No issue

-

### CODE REVIEW

File: packages/eset_protect/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.1.2"
```
Comment (efd6) [inline packages/eset_protect/changelog.yml:2]:
```suggestion
- version: "1.2.0"
```
We use semver, so enhancements bump minor, not patch.

File: packages/eset_protect/data_stream/detection/agent/stream/cel.yml.hbs [hunk 2]
```diff
@@ -61,6 +61,7 @@ program: |
         request.with({
           "Header":{
             "response-id": [state.cursor.response_id],
+            "3rd-integration": "Elastic",
```
Comment (efd6) [inline packages/eset_protect/data_stream/detection/agent/stream/cel.yml.hbs:64]:
```suggestion
            "3rd-integration": ["Elastic"],
```

File: packages/eset_protect/data_stream/device_task/agent/stream/cel.yml.hbs [hunk 3]
```diff
@@ -41,6 +41,7 @@ program: |
         request.with({
           "Header":{
             "response-id": [state.cursor.response_id],
+            "3rd-integration": "Elastic",
```
Comment (efd6) [inline packages/eset_protect/data_stream/device_task/agent/stream/cel.yml.hbs:44]:
```suggestion
            "3rd-integration": ["Elastic"],
```

**Review by efd6 — COMMENTED** [review]
Is there an issue that explains why this is happening?

Please add a proposed commit message to the PR description explaining what is being done and why.

The manifest.yml will need a change in the [`version`](https://github.com/elastic/integrations/blob/c55e51d9c8cdc64e4e10ca9c176894a74855d5a4/packages/eset_protect/manifest.yml#L4) field to match the entry in the changelog.

**Comment by efd6** [comment]
Is there documentation for this in the API docs?

**Comment by eset-enterprise-integration** [comment]
> Is there documentation for this in the API docs?

not yet, we will add it after release

**Comment by efd6** [comment]
Ideally, this should go in the commit message.

**Comment by eset-enterprise-integration** [comment]
> /test
we will see changes in our monitoring after release

**Comment by efd6** [comment]
Sorry, let me clarify: the link to documentation should go in the commit message. If it cannot, I will rewrite the commit message to explain that. This is less than ideal though.

**Comment by eset-enterprise-integration** [comment]
I add link to commit message. But information about header will be there later. Thanks

**Comment by efd6** [comment]
Thanks. Out of curiosity, is there a reason for not using [RFC 7231 5.5.3](https://www.rfc-editor.org/rfc/rfc7231.html#page-46) for this? This is well within what that was designed for?

**Review by efd6 — APPROVED** [review]

**Comment by eset-enterprise-integration** [comment]
> Thanks. Out of curiosity, is there a reason for not using [RFC 7231 5.5.3](https://www.rfc-editor.org/rfc/rfc7231.html#page-46) for this? This is well within what that was designed for.

We didn't know this

**Pull request merged**